catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -184,6 +192,10 @@ public void addFields(Metacard metacard, SolrInputDocument solrInputDocument)
             if (metacard.getAttribute(ad.getName()) != null) {
                 List<Serializable> attributeValues = metacard.getAttribute(ad.getName())
                         .getValues();
+                if (ad.getType()
```

Comment:
I addressed this in the commit message. 

> Dynamic schema resolver had trouble finding validation-warnings and
validation-errors fields in the fieldsCache. Now it doesn't. However
this may have been a fluke.

After I changed this and rebuilt it worked, but I'm not sure if it working was caused by this change or a different configuration. I think this portion may need more investigation.

There are two problems that make records not get removed. 1. The paging and batchsize didn't match up. 2. Before with `catalog-plugin-metacard-validation` on and now by default the query used to delete metacards will search for `validation-errors` and `validation-warnings`

The first problem _is_ fixed by the first commit and the second commit is _intended_ to fix the second.